pkg/reconciler/route/route.go

```diff
@@ -480,6 +447,71 @@ func (c *Reconciler) ensureFinalizer(route *v1alpha1.Route) error {
 	return err
 }
 
+func (c *Reconciler) updateRouteStatusURL(ctx context.Context, route *v1alpha1.Route) error {
+	_, existingPrivateServiceNames, _, desiredClusterLocalServiceNames, err := c.getServiceNames(ctx, route)
+	if err != nil {
+		return err
+	}
+
+	mainRouteServiceName, err := domains.HostnameFromTemplate(ctx, route.Name, "")
+	if err != nil {
+		return err
+	}
+
+	mainRouteMeta := route.ObjectMeta.DeepCopy()
+	allClusterLocalServiceNames := existingPrivateServiceNames.Union(desiredClusterLocalServiceNames)
+	labels.SetVisibility(mainRouteMeta, allClusterLocalServiceNames.Has(mainRouteServiceName))
+
+	host, err := domains.DomainNameFromTemplate(ctx, *mainRouteMeta, route.Name)
+	if err != nil {
+		return err
+	}
+
+	route.Status.URL = &apis.URL{
+		Scheme: "http",
+		Host:   host,
+	}
+
+	return nil
+}
+
+func (c *Reconciler) getServiceNames(ctx context.Context, route *v1alpha1.Route) (
+	existingPublicServiceNames sets.String,
+	existingClusterLocalServiceNames sets.String,
+	desiredPublicServiceNames sets.String,
+	desiredClusterLocalServiceNames sets.String,
+	err error,
+) {
+	// Populate existing service name sets
+	existingServices, err := c.getServices(route)
+	if err != nil {
+		return
```

Comment:
we're swallowing errors, which is not nice.

Reply:
This shouldn't be swallowing the error, as `existingServices` is a new variable while `err` is part of the named returns.

Reply:
Oh, god, I thought that 5 element list is argument list :(
Can we do a struct of sorts?

Reply:
@vagababov  I've wrapped the return in a struct, LMK what you think!